airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -186,6 +185,7 @@ def _read_incremental(
             for record_counter, record_data in enumerate(records, start=1):
                 yield self._as_airbyte_record(stream_name, record_data)
                 stream_state = stream_instance.get_updated_state(stream_state, record_data)
+                checkpoint_interval = stream_instance.state_checkpoint_interval
```

Comment:
this fixes issue with dynamic behaviour of state_checkpoint_interval (may change after read call)